File: java-receiver/src/test/java/org/zcode/javareceiver/util/ZcodeIdentifyCommand.java
package org.zcode.javareceiver.util;

import org.zcode.javareceiver.ZcodeLockSet;
import org.zcode.javareceiver.ZcodeOutStream;
import org.zcode.javareceiver.ZcodeResponseStatus;
import org.zcode.javareceiver.commands.ZcodeCommand;
import org.zcode.javareceiver.parsing.ZcodeCommandSequence;
import org.zcode.javareceiver.parsing.ZcodeCommandSlot;

public class ZcodeIdentifyCommand implements ZcodeCommand {

    @Override
    public void execute(final ZcodeCommandSlot slot, final ZcodeCommandSequence sequence, final ZcodeOutStream out) {
        out.writeField('V', (byte) 0)
                .continueField((byte) 0)
                .continueField((byte) 1)
                .writeBigStringField("Java/Test")
                .writeStatus(ZcodeResponseStatus.OK);
        slot.setComplete(true);
    }

    @Override
    public void finish(final ZcodeCommandSlot slot, final ZcodeOutStream out) {
    }

    @Override
    public void setLocks(final ZcodeLockSet locks) {
    }

    @Override
    public byte getCode() {
        return 0;
    }

    @Override
    public boolean matchesCode(final byte[] code) {
        return code.length == 1 && code[0] == getCode();
    }

    @Override
    public int getCodeLength() {
        return 1;
    }

    @Override
    public byte[] getFullCode() {
        return new byte[] { getCode() };
    }

}

File: receivers/native/arm/no-os/src/main/c++/commands/ZcodeIdentifyCommand.cpp
/*
 * ZcodeEchoCommand.cpp
 *
 *  Created on: 9 Sep 2020
 *      Author: robert
 */

#include "ZcodeIdentifyCommand.hpp"

void ZcodeIdentifyCommand::execute(ZcodeCommandSlot<ZcodeParameters> *slot, ZcodeCommandSequence<ZcodeParameters> *sequence, ZcodeOutStream<ZcodeParameters> *out) {
    out->writeStatus(OK);
    out->writeField('V', (uint8_t) 0);
    out->continueField((uint8_t) 0);
    out->continueField((uint8_t) 1);
    out->writeBigStringField("zcode/arm/no-os/initial-demo");
    slot->setComplete(true);
}
